docs/serving/using-knative-services.md

```diff
@@ -0,0 +1,233 @@
+address created by Knative.
+
+**NOTE:** If your cluster is new, it can take some time before the Service is
+assigned an external IP address.
+
+1. To find the IP address for your Service, enter:
+
+   ```shell
+   # In Knative 0.2.x and prior versions, the `knative-ingressgateway` service was used instead of `istio-ingressgateway`.
+   INGRESSGATEWAY=knative-ingressgateway
+
+   # The use of `knative-ingressgateway` is deprecated in Knative v0.3.x.
+   # Use `istio-ingressgateway` instead, since `knative-ingressgateway`
+   # will be removed in Knative v0.4.
+   if kubectl get configmap config-istio -n knative-serving &> /dev/null; then
+       INGRESSGATEWAY=istio-ingressgateway
+   fi
+
+   kubectl get svc $INGRESSGATEWAY --namespace istio-system
+   ```
+
+The command will return something similar to this:
+
+```shell
+   NAME                     TYPE           CLUSTER-IP     EXTERNAL-IP      PORT(S)                                      AGE
+   istio-ingressgateway   LoadBalancer   10.23.247.74   35.203.155.229   80:32380/TCP,443:32390/TCP,32400:32400/TCP   2d
+````
+
+Take note of the `EXTERNAL-IP` address.
+
+You can also export the IP address as a variable with the following command:
+
+```shell
+   export IP_ADDRESS=$(kubectl get svc $INGRESSGATEWAY --namespace istio-system --output 'jsonpath={.status.loadBalancer.ingress[0].ip}')
+```
+
+> Note: If you use minikube or a baremetal cluster that has no external load
+> balancer, the `EXTERNAL-IP` field is shown as `<pending>`. You need to use
+> `NodeIP` and `NodePort` to interact your app instead. To get your app's
+> `NodeIP` and `NodePort`, enter the following command:
+
+```shell
+   export IP_ADDRESS=$(kubectl get node  --output 'jsonpath={.items[0].status.addresses[0].address}'):$(kubectl get svc $INGRESSGATEWAY --namespace istio-system   --output 'jsonpath={.spec.ports[?(@.port==80)].nodePort}')
+```
+
+1. To find the host URL for your service, enter:
+
+   ```shell
+   kubectl get route helloworld-go  --output=custom-columns=NAME:.metadata.name,URL:.status.url
+   ```
+
+   The command will return the following:
+
+   ```shell
+   NAME                URL
+   helloworld-go       http://helloworld-go.default.example.com
+   ```
+
+   > **NOTE:** By default, Knative uses the `example.com` domain. To configure a
+   > custom DNS domain, see
+   > [Using a Custom Domain](../serving/using-a-custom-domain.md).
+
+   If you changed the name from `helloworld-go` to something else when creating
+   the `.yaml` file, replace `helloworld-go` in the above commands with the name
+   you entered.
+
+1. Now you can make a request to your app and see the results. Replace
+   `IP_ADDRESS` with the `EXTERNAL-IP` you wrote down, and replace
+   `helloworld-go.default.example.com` with the domain returned in the previous
+   step.
+
+   ```shell
+   curl -H "Host: helloworld-go.default.example.com" http://${IP_ADDRESS}
```

Comment:
It is probably worth explaining why we have to inject the host header here and that this is not necessary if you configure a domain associated with your IP.